av1an-core/src/broker.rs

```diff
@@ -269,52 +269,60 @@ impl<'a> Broker<'a> {
             tpl_crash_workaround = true;
           }
         } else {
+          if current_pass == passes {
+            let encoded_frames = ffmpeg::num_frames(chunk.output().as_ref()).unwrap();
+
+            if encoded_frames == chunk.frames {
+              let enc_time = st_time.elapsed();
+              let fps = encoded_frames as f64 / enc_time.as_secs_f64();
+
+              let progress_file = Path::new(&self.project.temp).join("done.json");
+              get_done().done.insert(
+                chunk.name(),
+                DoneChunk {
+                  frames: encoded_frames,
+                  size_bytes: Path::new(&chunk.output())
+                    .metadata()
+                    .expect("Unable to get size of finished chunk")
+                    .len(),
+                },
+              );
+
+              let mut progress_file = File::create(&progress_file).unwrap();
+              progress_file
+                .write_all(serde_json::to_string(get_done()).unwrap().as_bytes())
+                .unwrap();
+
+              update_progress_bar_estimates(
+                frame_rate,
+                self.project.frames,
+                self.project.verbosity,
+                audio_size_bytes.load(atomic::Ordering::SeqCst),
+              );
+
+              debug!(
+                "finished chunk {:05}: {} frames, {:.2} fps, took {:.2?}",
+                chunk.index, chunk.frames, fps, enc_time
+              );
+            } else {
+              warn!(
+                "finished chunk: FRAME MISMATCH: chunk {}: {}/{} (actual/expected frames)",
```

Comment:
I think we also have to explicitly return an error from this entire function if we are on the last try, similar to what we do on line 253, since otherwise I think this function will actually not return any type of error at all if the frame count mismatch happens `max_tries` times.

Reply:
In order to avoid a lot of this duplication, I moved the error handling logic up into `create_pipes` so we'll just get an Err back from that function and it'll be handled by the existing code here.